Preprocessor: parentheses around `defined` in preprocessor cause incorrect mismatched parens error

```c
#if (defined(FOO))
#endif
```
```
./test.c:1:7: error: expected ')' before end of file
#if (defined(FOO))
      ^
./test.c:1:5: note: to match this '('
#if (defined(FOO))
    ^
1 error generated.
```

--- a/src/Preprocessor.zig
+++ b/src/Preprocessor.zig
@@ -628,7 +628,7 @@ fn expr(pp: *Preprocessor, tokenizer: *Tokenizer) MacroError!bool {
             else => if (tok.id.isMacroIdentifier()) {
                 if (tok.id == .keyword_defined) {
                     const tokens_consumed = try pp.handleKeywordDefined(&tok, items[i + 1 ..], eof);
-                    i += tokens_consumed + 1; // + 1 for keyword_defined
+                    i += tokens_consumed;
                 } else {
                     try pp.comp.diag.add(.{
                         .tag = .undefined_macro,

--- a/test/cases/macro expansion to defined.c
+++ b/test/cases/macro expansion to defined.c
@@ -44,6 +44,12 @@
 #error Foo should be defined
 #endif
 
+#if defined(NOT_DEFINED)
+#error NOT_DEFINED should not be defined
+#elif !DEFINED ( FOO )
+#error FOO should be defined
+#endif
+
 #define EXPECTED_ERRORS \
 	"macro expansion to defined.c:5:5: warning: macro expansion producing 'defined' has undefined behavior [-Wexpansion-to-defined]" \
 	"macro expansion to defined.c:3:24: note: expanded from here" \

